google/cloud/firestore_v1/async_aggregation.py

```diff
@@ -0,0 +1,122 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Classes for representing Async aggregation queries for the Google Cloud Firestore API.
+
+A :class:`~google.cloud.firestore_v1.async_aggregation.AsyncAggregationQuery` can be created directly from
+a :class:`~google.cloud.firestore_v1.async_collection.AsyncCollection` and that can be
+a more common way to create an aggregation query than direct usage of the constructor.
+"""
+from __future__ import annotations
+
+from google.api_core import gapic_v1
+from google.api_core import retry as retries
+
+from typing import List, Union
+
+
+from google.cloud.firestore_v1.base_aggregation import (
+    AggregationResult,
+    _query_response_to_result,
+    BaseAggregationQuery,
+)
+
+
+class AsyncAggregationQuery(BaseAggregationQuery):
+    """Represents an aggregation query to the Firestore API."""
+
+    def __init__(
+        self,
+        nested_query,
+    ) -> None:
+        super(AsyncAggregationQuery, self).__init__(nested_query)
+
+    async def get(
+        self,
+        transaction=None,
+        retry: Union[
+            retries.Retry, None, gapic_v1.method._MethodDefault
+        ] = gapic_v1.method.DEFAULT,
+        timeout: float | None = None,
+    ) -> List[AggregationResult]:
+        """Runs the aggregation query.
+
+        This sends a ``RunAggregationQuery`` RPC and returns a list of aggregation results in the stream of ``RunAggregationQueryResponse`` messages.
+
+        Args:
+            transaction
+                (Optional[:class:`~google.cloud.firestore_v1.transaction.Transaction`]):
+                An existing transaction that this query will run in.
+                If a ``transaction`` is used and it already has write operations
+                added, this method cannot be used (i.e. read-after-write is not
+                allowed).
+            retry (google.api_core.retry.Retry): Designation of what errors, if any,
+                should be retried.  Defaults to a system-specified policy.
+            timeout (float): The timeout for this request.  Defaults to a
+                system-specified value.
+
+        Returns:
+            list: The aggregation query results
+
+        """
+        result = self.stream(transaction=transaction, retry=retry, timeout=timeout)
+        result = [aggregation async for aggregation in result]
+        return result
+
+    async def stream(
+        self,
+        transaction=None,
+        retry: Union[
+            retries.Retry, None, gapic_v1.method._MethodDefault
+        ] = gapic_v1.method.DEFAULT,
+        timeout: float | None = None,
+    ):  # -> Generator[document.DocumentSnapshot, Any, None]: AsyncGenerator
```

Comment:
should this be commented out?

Reply:
It actually should be uncommented so we have type annotations. I have fixed this.